### PR TITLE
added chunk_every function

### PR DESCRIPTION
**Pull request by basbl**
## chunk_every

By passing a `$size`, `$step` and whether to `$discardIncompleteChunk` you have full control over the chunks created.

```php
chunk_every([1, 2, 3], 2); // [[1, 2], [2, 3], [3]]
chunk_every([1, 2, 3], 2, 1, true); // [[1, 2], [2, 3]]
```

Inspired by [Elixir](https://hexdocs.pm/elixir/Enum.html#chunk_every/4)

### CODE REVIEW

File: src/Functional/ChunkEvery.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * @param int $step Step with which to increment start of next chunk
+ * @param bool $discardIncompleteChunk Discard the last chunk if it does not have the specified size.
+ * @return array
+ */
+function chunk_every($collection, $size = 1, $step = 1, $discardIncompleteChunk = false)
+{
+    InvalidArgumentException::assertCollection($collection, __FUNCTION__, 1);
+
+    if ($collection instanceof Traversable) {
+        $array = \iterator_to_array($collection);
```
Comment (lstrojny) [inline src/Functional/ChunkEvery.php]: This won’t work with an infinite iterator or an iterator where the keys are duplicated

**Review by lstrojny — CHANGES_REQUESTED** [review]

File: src/Functional/ChunkEvery.php (hunk 1 above)
Comment (basbl) [reply src/Functional/ChunkEvery.php]: You're right, I'll tinker some more!

Comment (basbl) [reply src/Functional/ChunkEvery.php]: What kind of solution would you prefer for the InfiniteIterator case? Is reducing it to its inner iterator acceptable?

**Comment by lstrojny** [comment]
Closing because of inactivity

**Pull request closed without merging**